dds/idl/dds_generator.h

```diff
@@ -505,6 +503,23 @@ std::ostream& operator<<(std::ostream& o,
   }
 }
 
+inline std::string bounded_arg(AST_Type* type)
+{
+  using namespace AstTypeClassification;
+  std::ostringstream arg;
+  const Classification cls = classify(type);
+  if (cls & CL_STRING) {
+    AST_String* const str = AST_String::narrow_from_decl(type);
```

Comment:
use dynamic_cast instead of narrow_from_decl

Reply:
That's not part of my PR I can really change since it's coming from #1812, where [it's already been changed](https://github.com/objectcomputing/OpenDDS/pull/1812/files/b0d4fae835cc5172fa8b07e5060042f19f67a387..364675a384b87d48db01aac1962dcfbb864018bf). I just need to update this PR.